src/components/molecules/VenueJoinButton/VenueJoinButton.tsx

```diff
@@ -0,0 +1,180 @@
+import React, { useCallback, useState } from "react";
+
+import { Modal } from "react-bootstrap";
+import { useForm } from "react-hook-form";
+
+import {
+  DEFAULT_PARTY_BUTTON_TEXT,
+  DEFAULT_PARTY_JOIN_MESSAGE,
+} from "settings";
+
+import { checkAccess } from "api/auth";
+
+import { AnyVenue } from "types/venues";
+
+import { useUser } from "hooks/useUser";
+
+import { getCurrentTimeInUTCSeconds, getTimeBeforeParty } from "utils/time";
+import { setLocalStorageToken } from "utils/localStorage";
+import { isTruthy } from "utils/types";
+import { joinVenue } from "utils/url";
+
+import "./VenueJoinButton.scss";
+
+export interface VenueJoinButtonProps {
+  venueId?: string;
+  venue: AnyVenue;
```

Comment:
Can we do this?

```suggestion
  venue: WithId<AnyVenue>;
```

Reply:
If so, there is a `withId` helper that can be used on the calling side to construct the appropriately typed object.